presto-function-namespace-managers/src/main/java/com/facebook/presto/functionNamespace/db/FunctionNamespaceDao.java

```diff
@@ -0,0 +1,240 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.functionNamespace.db;
+
+import com.facebook.presto.spi.function.RoutineCharacteristics.Language;
+import com.facebook.presto.spi.function.SqlFunctionId;
+import com.facebook.presto.spi.function.SqlInvokedFunction;
+import com.facebook.presto.spi.function.SqlParameter;
+import com.facebook.presto.spi.type.TypeSignature;
+import org.jdbi.v3.sqlobject.config.RegisterArgumentFactories;
+import org.jdbi.v3.sqlobject.config.RegisterArgumentFactory;
+import org.jdbi.v3.sqlobject.config.RegisterRowMapper;
+import org.jdbi.v3.sqlobject.config.RegisterRowMappers;
+import org.jdbi.v3.sqlobject.customizer.Bind;
+import org.jdbi.v3.sqlobject.customizer.Define;
+import org.jdbi.v3.sqlobject.statement.SqlQuery;
+import org.jdbi.v3.sqlobject.statement.SqlUpdate;
+
+import java.util.List;
+import java.util.Optional;
+
+@RegisterRowMappers({
+        @RegisterRowMapper(SqlInvokedFunctionRowMapper.class),
+        @RegisterRowMapper(SqlInvokedFunctionRecordRowMapper.class)
+})
+@RegisterArgumentFactories({
+        @RegisterArgumentFactory(SqlFunctionIdArgumentFactory.class),
+        @RegisterArgumentFactory(SqlParametersArgumentFactory.class),
+        @RegisterArgumentFactory(TypeSignatureArgumentFactory.class),
+})
+public interface FunctionNamespaceDao
+{
+    @SqlUpdate("CREATE TABLE IF NOT EXISTS <table_name> (\n" +
+            "   catalog_name varchar(128) NOT NULL,\n" +
+            "   schema_name varchar(128) NOT NULL,\n" +
+            " PRIMARY KEY (catalog_name, schema_name)\n" +
+            ")")
+    void createFunctionNamespacesTableIfNotExists(@Define("table_name") String tableName);
+
+    @SqlUpdate("CREATE TABLE IF NOT EXISTS <table_name> (\n" +
+            "  id bigint(20) NOT NULL AUTO_INCREMENT,\n" +
+            "  function_id varchar(700) NOT NULL,\n" +
+            "  version bigint(20) unsigned NOT NULL,\n" +
+            "  catalog_name varchar(128) NOT NULL,\n" +
+            "  schema_name varchar(128) NOT NULL,\n" +
+            "  function_name varchar(256) NOT NULL,\n" +
+            "  parameters text NOT NULL,\n" +
+            "  return_type varchar(256) NOT NULL,\n" +
+            "  language varchar(64) NOT NULL,\n" +
+            "  `deterministic` tinyint(1) NOT NULL,\n" +
```

Comment:
I think it's more descriptive to use `boolean` rather than tinyint(1). Tinyint(1) is how MySQL implements boolean but that's its implementation detail.